File: stone_py/user_prog.py
#!/usr/bin/env python
"""user_prog.py to demonstrate the Programming API of stone.py"""

import stone

# dummy try
r = stone.brick_weight(1,2,3, 1)
print(r)

# mass of a limestone brick
r = stone.brick_weight(density=2800)
print("limestone brick mass: {:.1f} kg".format(r))

# thermal conductivity of a limestone brick
r = stone.wall_thermal_conductivity(conductivity=3.1, thickness=0.2)
print("thermal conductivity of a limestone wall: {:.1f} W/(m2.K)\n".format(r))

# display the call statistics
r = stone.call_activities()
#print(r)
print(r['visit_stat'])

File: stone_js/user_prog.js
// user_prog.js

'use strict';

// A black-magic solution from ESM, waiting for the official support for es6-module by nodejs, planned with node-V12.0
require = require("esm")(module/*, options*/);
const stone = require("./web_ui/js/stone.js");
// With node V12.0, this will be replaced by
//import * as stone from './web_ui/js/stone.js';

// dummy try
var r = stone.stoneWeight(1,2,3, 1);
console.log("stone.stoneWeight(1,2,3, 1): " + r);

// mass of a limestone brick
var r = stone.stoneWeight(0.4, 0.2, 0.2, 2800);
//var r = stone.stoneWeight(0.4, 0.2, 0.2); // thanks to paramter default values
//var r = stone.stoneWeight(); // thanks to paramter default values
console.log("limestone brick mass: " + r.toFixed(2) + " kg");

// thermal conductivity of a limestone brick
var r = stone.wallThermalConductivity(3.1, 0.2);
console.log("thermal conductivity of limeston wall: " + r.toFixed(1) + " W/(m2.K)\n");

// display the call statistics
var r = stone.callActivities();
console.log("Print a part of the json:");
console.log(r.visit_stat);

console.log("end of user_prog.js");

